src/radical/pilot/agent/launch_method/mpiexec.py

```diff
@@ -76,15 +77,28 @@ def _init_from_scratch(self, env, env_sh):
             lm_info['mpt']     = True
 
         # check that this implementation allows to use `rankfile` option
-        lm_info['use_rf'] = bool(ru.sh_callout('%s --help |& grep -- "-rf"' %
-                                               lm_info['command'])[0])
+        lm_info['use_rf'] = self._check_rank_host_file(lm_info['command'],
+                                                       '-rf')
+
+        # if we fail, then check if this implementation allows to use
+        # `host names` option
+        if not lm_info['use_rf']:
+            lm_info['use_hf'] = self._check_rank_host_file(lm_info['command'],
+                                                           '-f')
 
         mpi_version, mpi_flavor = self._get_mpi_info(lm_info['command'])
         lm_info['mpi_version']  = mpi_version
         lm_info['mpi_flavor']   = mpi_flavor
 
         return lm_info
 
+    # --------------------------------------------------------------------------
+    #
+    def _check_rank_host_file(self, lm_cmd, param):
```

Comment:
can you please call this method something like `_lm_option_availability` (since this is a general check method)